Source/WebCore/Modules/fetch/FetchHeaders.cpp

```diff
@@ -229,15 +269,20 @@ std::optional<KeyValuePair<String, String>> FetchHeaders::Iterator::next()
 {
     if (m_keys.isEmpty() || m_updateCounter != m_headers->m_updateCounter) {
         m_keys.resize(0);
-        m_keys.reserveCapacity(m_headers->m_headers.size());
+        m_keys.reserveCapacity(m_headers->m_headers.size() + m_headers->m_setCookieValues.size());
         for (auto& header : m_headers->m_headers)
-            m_keys.uncheckedAppend(header.key.convertToASCIILowercase());
-        std::sort(m_keys.begin(), m_keys.end(), WTF::codePointCompareLessThan);
+            m_keys.uncheckedAppend(std::pair(header.key.convertToASCIILowercase(), 0));
+        for (size_t i = 0; i < m_headers->m_setCookieValues.size(); i++)
+            m_keys.uncheckedAppend(std::pair("set-cookie"_str, i));
+        std::sort(m_keys.begin(), m_keys.end(), [](std::pair<String, size_t> a, std::pair<String, size_t> b) {
+            return WTF::codePointCompareLessThan(a.first, b.first) || (a.first == b.first && a.second < b.second);
+        });
         m_updateCounter = m_headers->m_updateCounter;
     }
     while (m_currentIndex < m_keys.size()) {
-        auto key = m_keys[m_currentIndex++];
-        auto value = m_headers->m_headers.get(key);
+        auto pair = m_keys[m_currentIndex++];
+        String key(WTFMove(pair.first));
+        auto value = (key == "set-cookie"_str) ? m_headers->m_setCookieValues[pair.second] : m_headers->m_headers.get(key);
```

Comment:
We probably do not need to have a Vector<String, size_t> here.
We could simply keep a m_setCoookieCounter in the Iterator and when String is set_cookie, increment it/

Reply:
That wouldn't work when the headers are modified in the middle of the iteration. But if we also store the current size of `m_setCookieValues` in the iterator, this could be worked around in the invalidation recovery code in `next()`. I'll give that a try.

Reply:
I changed this to use a `m_setCookieIndex` rather than pairs. As it turns out, keeping the previous size of `m_setCookieValues` wasn't needed. Please let me know if this needs some more documentation.